### PR TITLE
chore(FDS-189):[Cascara]: Validate data-component usage on Cascara 

### PR DESCRIPTION
**Pull request by rociobaena**
Validate `data-component` usage on Cascara components

 Add the identifiable `data-component="ComponentName"`  to our components in Cascara.

### Snapshots
The new snapshots should reflect the data-component name.

## PR Author Checklist

- [x] PR title adheres to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#specification)
- [x] I have finished reviewing the contents of this PR for accuracy
- [x] If needed, I have included a [Changeset](https://github.com/changesets/changesets) and it follows [Semantic Versioning principles](https://semver.org/)
- [x] I have completed all of the above and have enabled `auto-merge` for this PR

### CODE REVIEW

**Review by lordpixel — APPROVED** [review]

**Comment by lordpixel** [comment]
@rociobaena this looks good, we only need to update snapshots.

**Review by lordpixel — DISMISSED** [review]

File: packages/cascara/src/ui/Columns/Columns.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ const propTypes = {
   style: pt.shape(pt.object),
 };
 
+const componentName = 'Columns';
+
```
Comment (brianespinosa) [inline packages/cascara/src/ui/Columns/Columns.js:25]: I don't see anywhere else in the component where we are using the `componentName` const. To me this feels like extra work for something that we should not be considering a configurable global. This data attribute should always be the name of the component itself. I think we should just put these strings directly on the `data-component='Columns'` attribute.

File: packages/cascara/src/ui/List/List.js [hunk 2]
```diff
@@ -9,10 +9,17 @@ const propTypes = {
...
 
+const componentName = 'List';
 const List = ({ data, ...rest }) => {
   const compositeState = useCompositeState();
   return (
-    <Composite {...rest} className={styles._} {...compositeState} as='ul'>
+    <Composite
+      {...rest}
+      className={styles._}
+      {...compositeState}
```
Comment (brianespinosa) [inline packages/cascara/src/ui/List/List.js:19]:
```suggestion
      {...compositeState}
      className={styles._}
```
Let's move this spread value above className

**Review by brianespinosa — CHANGES_REQUESTED** [review]
@rociobaena I am seeing that we are defining componentName as a const above the component and using that on the component below. Because are not using componentName anywhere else, this is extra abstraction that I do not think helps us right now. Let's go ahead and define this string directly on the data-component attribute.

**Comment by rociobaena** [comment]
I close this PR due to a rename this branch to include a description, this includes the feedback provided here: https://github.com/Espressive/cascara/pull/345.

**Pull request closed without merging**